### ISSUE TITLE
select vs sel

### ISSUE THREAD

**Issue by d-chambers** [DASDAE/dascore#11]
**Description**
slack message from Jin:

It seems that the select function cannot take a list of sample locations.

![image](https://user-images.githubusercontent.com/11671536/147839828-1d7c2a5a-f5df-4235-a022-0d775c1120a8.png)

**Comment by d-chambers** [comment]
@jinwar,

This is indeed a bug. However, `select` should raise an error when passed an array of sample locations. I don't want to support this behavior because of how easily it breaks the assumption of even sampling in each dimension. Consider the following example:

```python
import numpy as np

import dascore as dc
# get example patch
patch = dc.get_example_patch()

# load time array, take irregular chunks of it.
time = dc.coords['time']
time_indices = np.array([1,2,3,5,12,40)
new_time = time[time_indices]

# now calling `patch.select` with new_time will create an unevenly sampled patch in the time dimension
patch_uneven = patch.select(time=new_time)
 
# However, if we restrict `select` to only a tuple of ranges (start, stop), it cannot change the sampling interval
patch_event = patch.select(time=(time[0], time[10]) )
```

It's not that we *cant* support patches with uneven sampling rates, it will just require more work since they break many of the patch methods/transforms (eg fft, pass_filter).

Moreover, If we supported both styles it introduces ambiguity with 2 length sequences. For example:

```python
patch.select(time=(time[0], time[10])) 
```
Might mean return a patch with only time slices from time[0] and time[10], or it might mean return a range for all time samples between time[0] and time[10]. Of course, we could go the numpy route where different objects have different meanings (eg, tuples vs lists vs arrays) but now it is getting very complicated.

**Comment by d-chambers** [comment]
I think there should be an `iselect` method which simply works on ranges of indices though, which may alleviate or remove the need for using arrays in `select`. 

```python
import dascore as dc

patch = dc.get_example_patch()

# slice patch along first 10 samples
out = patch.iselect(time=(0, 10)) 
```

**Comment by jinwar** [comment]
The function I would like to have is a resampling of the data at certain time/spacing points. I added a set and iset methods just to pass the arguments to the xarray.sel and xarray.iset. We may want to think what we can implement these. 

Here is what I have for my low-frequency processing package:

            lfDAS = DASdata.pass_filter(time=(None,1/dt/2*0.9))\
                            .sel(time=time_grid[data_end-buff_size:new_data_end-buff_size]
                                    ,method='nearest')

This will also allow increasing the sampling rate.

**Comment by d-chambers** [comment]
I am thinking this can still work, provided we check that input arrays are evening sampled, otherwise we need to set `d_{dim}` to `NaN` so that functions that depend on evenly sampled dimensions know the data aren't evenly sampled.

**Comment by d-chambers** [comment]
Closed in favor of interpolation methods implemented in #19
